feature/platform/tests/breakout_configuration/breakout_base_test.go

```diff
@@ -0,0 +1,232 @@
+		newIP4 := (ip4 | uint32(lastOctet)) + (uint32(i) << 8)
+		incrementedIP := make(net.IP, 4)
+		binary.BigEndian.PutUint32(incrementedIP, newIP4)
+		ips[i] = fmt.Sprintf("%s/%d", incrementedIP.String(), 24) // Format as a string with a /24 subnet.
+	}
+
+	return ips, nil
+}
+
+// findNewPortNames fetches the parent breakout port interface states and the new port convention.
+// Example being 4x100 parent port would be FourHundredGigE0/0/0/10 this will find and return
+// the newly broken out ports of OneHundredGigE0/0/0/0/10/0-4
+func findNewPortNames(dut *ondatra.DUTDevice, t *testing.T, originalPortName string, numBreakouts uint8) ([]string, error) {
+	// Fetch the current state of all interfaces from the device using gNMI.
+	intfs := gnmi.Get(t, dut, gnmi.OC().InterfaceMap().State())
+
+	// Split the original port name by '/' to extract the correct index (third-last segment in this case).
+	portSegments := strings.Split(originalPortName, "/")
+	if len(portSegments) < 4 {
+		return nil, fmt.Errorf("invalid port name format: %v", originalPortName)
+	}
+	portIndex := portSegments[len(portSegments)-2] // Get the third-last segment, which is "30"
+
+	// Define a pattern to match breakout port names that include the original port index.
+	breakoutPattern := fmt.Sprintf(`\w+/\d+/\d+/%s/\d+`, portIndex)
+
+	// Compile the pattern into a regular expression.
+	re := regexp.MustCompile(breakoutPattern)
+
+	// Loop through all interfaces and collect those that match the breakout pattern.
+	newPortNames := []string{}
+	for intfName := range intfs {
+		if re.MatchString(intfName) {
+			newPortNames = append(newPortNames, intfName)
+		}
+	}
+
+	// Check if the number of new ports found is equal to the number of breakouts expected.
+	if len(newPortNames) != int(numBreakouts) {
+		return nil, fmt.Errorf("expected to find %d new ports, found %d", numBreakouts, len(newPortNames))
+	}
+
+	return newPortNames, nil
+}
+
+// fetchResponses will fetch the ping response
+func fetchResponses(c spb.System_PingClient) ([]*spb.PingResponse, error) {
+	pingResp := []*spb.PingResponse{}
+	for {
+		resp, err := c.Recv()
+		switch {
+		case err == io.EOF:
+			return pingResp, nil
+		case err != nil:
+			return nil, err
+		default:
+			pingResp = append(pingResp, resp)
+		}
+	}
+}
+
+func sortBreakoutPorts(breakOutPorts []string) {
+	sort.Slice(breakOutPorts, func(i, j int) bool {
+		// Extract the trailing number after the last "/"
+		numberI, errI := strconv.Atoi(strings.Split(breakOutPorts[i], "/")[4])
+		numberJ, errJ := strconv.Atoi(strings.Split(breakOutPorts[j], "/")[4])
+		// Handle potential errors from Atoi conversion
+		if errI != nil || errJ != nil {
+			return false
+		}
+		return numberI < numberJ
+	})
+}
+
+// Function to get compName and set up port name based on provided interface and port prefix
+func getCompName(dut *ondatra.DUTDevice, string, portPrefix string, t *testing.T) (string, string, bool) {
```

Comment:
this whole function is vendor-specific.
if it is impossible to eliminate vendor specific logic and dependency on prefixes, then at least vendor-specific items should be guarded with switch/case statements and/or maps to select the correct choices for each vendor.

Reply:
It is expected that component/interface names are vendor specific.  Agreed, they should be wrapped/guarded by switch statement like switch dut.Vendor()

Reply:
>It is expected that component/interface names are vendor specific.

I agree. My comment is more generic: in this function, and throughout the test (I left a similar comment above, which was left unanswered), there are a lot of dependencies on hardcoded names/prefixes/etc, and I suspect that some of this logic can (and should) be replaced with dynamic discovery or other methods. 

For example, not all vendors use prefixes to distinguish interfaces with different port speeds; however the current logic seem to require that:

https://github.com/openconfig/featureprofiles/blob/676ca31db676d47dd127c1e8a5f75b7ddd1f0686/feature/platform/tests/breakout_configuration/breakout_test.go#L156-L189

Thus this logic needs to be revisited.

Reply:
these area's you have called out as vendor specify have been wrapped in if `dut.Vendor() == ondatra.CISCO {}` so only cisco devices will access that code block.